src/main/java/Parser.java

```diff
@@ -1,33 +1,123 @@
-import java.io.IOException;
+import java.util.DuplicateFormatFlagsException;
 
 public class Parser {
 
-    public static TaskType returnsTaskType (String command) throws DukeException {
+    public static Command parse (String command) throws DukeException {
 
         String[] words = command.split(" ", 2);
 
+        Task task = null;
+        int index;
 
         switch (words[0]) {
         case "todo":
-            return TaskType.TODO;
+            task = Parser.parsesAdd(command, TaskType.TODO);
+            return new AddCommand(task);
         case "deadline":
-            return TaskType.DEADLINE;
+            task = Parser.parsesAdd(command, TaskType.DEADLINE);
+            return new AddCommand(task);
         case "event":
-            return TaskType.EVENT;
+            task = Parser.parsesAdd(command, TaskType.EVENT);
+            return new AddCommand(task);
         case "done":
-            return TaskType.DONE;
+            index = Parser.parsesDone(command);
+            return new DoneCommand(index);
         case "delete":
-            return TaskType.DELETE;
+            index = Parser.parsesDone(command);
+            return new DeleteCommand(index);
         }
 
         switch (command) {
         case "list":
-            return Bot.TaskType.LIST;
+            return new ListCommand();
         case "bye":
-            return Bot.TaskType.BYE;
+            return new ExitCommand();
```

Comment:
Would it be better if you kept these two commands together with the rest of the switch statement so that it remains as a block? If you have no spaces, split actually returns you an array containing just the command itself, so `word[0]` would still correspond to "list" and "bye" in those cases.

Reply:
Yes, I do agree with Terence